File: modules/angular2/test/core/compiler/component_url_mapper_spec.js
import {describe, it, expect, beforeEach, ddescribe, iit, xit, el} from 'angular2/test_lib';

import {
  ComponentUrlMapper,
  RuntimeComponentUrlMapper
} from 'angular2/src/core/compiler/component_url_mapper';

export function main() {
  describe('RuntimeComponentUrlMapper', () => {
    it('should return the registered URL', () => {
      var url = 'http://path/to/component';
      var mapper = new RuntimeComponentUrlMapper();
      mapper.setComponentUrl(SomeComponent, url);
      expect(mapper.getUrl(SomeComponent)).toEqual(url);
    });

    it('should fallback to ComponentUrlMapper', () => {
      var mapper = new ComponentUrlMapper();
      var runtimeMapper = new RuntimeComponentUrlMapper();
      expect(runtimeMapper.getUrl(SomeComponent)).toEqual(mapper.getUrl(SomeComponent));
    });
  });
}

class SomeComponent {}

File: modules/angular2/test/core/compiler/component_url_mapper_spec.dart
library angular2.test.core.compiler.component_url_mapper_spec;

import 'package:angular2/test_lib.dart';
import 'package:angular2/src/core/compiler/component_url_mapper.dart';

main() {
  describe("ComponentUrlMapper", () {
    it("should return the URL of the component's library", () {
      var mapper = new ComponentUrlMapper();
      expect(mapper
          .getUrl(SomeComponent)
          .endsWith("core/compiler/component_url_mapper_spec.dart")).toBeTrue();
    });
  });
}

class SomeComponent {}
